native/android/src/main/java/com/nozbe/watermelondb/WMDatabase.java

```diff
@@ -0,0 +1,191 @@
+package com.nozbe.watermelondb;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteCursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+public class WMDatabase {
+    private final SQLiteDatabase db;
+
+    private WMDatabase(SQLiteDatabase db) {
+        this.db = db;
+    }
+
+    public static Map<String, WMDatabase> INSTANCES;
+
+    public static WMDatabase getInstance(String name, Context context, int openFlags) {
+        synchronized (WMDatabase.class) {
+            if (!INSTANCES.containsKey(name) ||
+                    !(INSTANCES.get(name) == null && INSTANCES.get(name).isOpen())) {
```

Comment:
this condition seems wrong

Reply:
Duh, of course it's wrong
It should be `!= null`

Reply:
ok, fixed on another branch